### PR TITLE
Disable nativeCheckoutPledgeView feature flag on release builds

### PR DESCRIPTION
**Pull request by ifbarrera**
# 📲 What

Disable the `nativeCheckoutPledgeView` feature flag on release builds.

# 🤔 Why

So that current release builds don't get an incomplete pledging experience once that feature flag is turned on in the future.

# ✅ Acceptance criteria

- [x] when running the app on debug mode, the `Native Checkout Pledge View` feature flag should function as normal - turning it on should show the new pledge screen, turning it off should show the old pledge screen
- [x] when running the app in `Release` mode, you should never see the new pledge screen experience.

### CODE REVIEW

File: Library/Extensions/Feature+Helpers.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@ public func featureNativeCheckoutIsEnabled() -> Bool {
 
 public func featureNativeCheckoutPledgeViewIsEnabled() -> Bool {
```
Comment (justinswart) [inline Library/Extensions/Feature+Helpers.swift]: I feel like a `guard` more clearly explains the 'gate-keeping' nature of what we want to do here:

```swift
public func featureNativeCheckoutPledgeViewIsEnabled() -> Bool {
  // native checkout pledge view is currently not available in release builds
  guard !AppEnvironment.current.mainBundle.isRelease else { return false }

  return Feature.nativeCheckoutPledgeView.isEnabled()
}
```

File: Library/Extensions/Feature+HelpersTests.swift [hunk 2]
```diff
@@ -56,4 +56,37 @@ final class FeatureHelpersTests: TestCase {
       XCTAssertFalse(featureNativeCheckoutPledgeViewIsEnabled())
     }
   }
+
+  func testFeatureNativeCheckoutPledgeViewEnabled_isFalse_ffIsFalse_inReleaseBuild() {
```
[the remark this answers is not in the document]
Comment (justinswart) [reply Library/Extensions/Feature+HelpersTests.swift]: fast forward
friendly fire
furry friend
flipping flipper
feature flag

[the remark this answers is not in the document]
Comment (ifbarrera) [reply Library/Extensions/Feature+HelpersTests.swift]: Bingo...

Feature Flag

**Comment by ifbarrera** [comment]
Hey @dusi, actually on `KickBeta`, `KickAlpha` and `Debug` the feature flags should both continue to work as normal. It's only on `Kickstarter` (Testflight or AppStore) that this would have an effect. We refer to the `Kickstarter` app as the "release" app, but really it just means "the app with the `com.kickstarter.kickstarter` bundle identifier".

The way to test this in `Release` mode is to change your build configuration to `Release` in `Edit Scheme` and run it on a device. You shouldn't be able to access the debug menu at all on `Release` mode, because it's the same as a Testflight build. Unfortunately you'd also have to be in the `native checkout` variant.

File: Library/Extensions/Feature+HelpersTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (dusi) [reply Library/Extensions/Feature+HelpersTests.swift]: From my Counter Strike days Friendly Fire was my best guess :trollface: Flipping flipper is also pretty good :D

**Comment by dusi** [comment]
I can verify that building for `Release` configuration I am not able to see the new pledge screen even though I'm taken to the new carousel.

But something still has to be broken since the feature toggles don't do anything for me on the Alpha build. Even though the toggles for `Native Checkout` and `Native Checkout Pledge Screen` are ON I'm still going through the old flow.

**Comment by justinswart** [comment]
> I can verify that building for `Release` configuration I am not able to see the new pledge screen even though I'm taken to the new carousel.
> 
> But something still has to be broken since the feature toggles don't do anything for me on the Alpha build. Even though the toggles for `Native Checkout` and `Native Checkout Pledge Screen` are ON I'm still going through the old flow.

Sometimes you need to background/foreground the app to get the config to sync after logging in 🤔

**Comment by dusi** [comment]
Yep, did that ... can you try with the latest Alpha build? Maybe I'm holding it wrong 😄

**Comment by justinswart** [comment]
Oh I see now. The effective code is behind `userCanSeeNativeCheckout()` in https://github.com/kickstarter/ios-oss/blob/master/Library/ProjectPamphlet+Helpers.swift - not sure why the file is named like that because it's not an extension.

This is used in `ProjectPamphletContentDataSource`. So we would need to do some additional work to make this work correctly. We could have this type of `guard` logic in `userCanSeeNativeCheckout()` too but I think then we'd be duplicating it across 3 places.

Let's give it some more thought and see how we can better-consolidate this.

**Comment by ifbarrera** [comment]
Okay I figured out what happened, it was the feature experiment stuff that was making the feature toggles useless. It's been fixed on `master` so all I had to do was pull, should be working now.

**Pull request merged**